core/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointTrackerTests.java

```diff
@@ -75,13 +74,28 @@ public void testEmptyShards() {
         return allocations;
     }
 
-    private static IndexShardRoutingTable routingTable(Set<AllocationId> initializingIds) {
+    private static IndexShardRoutingTable routingTable(final Set<AllocationId> initializingIds, final AllocationId primaryId) {
+        final ShardId shardId = new ShardId("test", "_na_", 0);
+        final ShardRouting primaryShard =
+                TestShardRouting.newShardRouting(shardId, randomAlphaOfLength(10), null, true, ShardRoutingState.STARTED, primaryId);
+        return routingTable(initializingIds, primaryShard);
+    }
+
+    private static IndexShardRoutingTable routingTable(final Set<AllocationId> initializingIds, final ShardRouting primaryShard) {
+        assert !initializingIds.contains(primaryShard.allocationId());
         ShardId shardId = new ShardId("test", "_na_", 0);
         IndexShardRoutingTable.Builder builder = new IndexShardRoutingTable.Builder(shardId);
         for (AllocationId initializingId : initializingIds) {
-            builder.addShard(TestShardRouting.newShardRouting(shardId, randomAlphaOfLength(10), null, false, ShardRoutingState.INITIALIZING,
-                initializingId));
+            builder.addShard(TestShardRouting.newShardRouting(
+                    shardId, randomAlphaOfLength(10), null, false, ShardRoutingState.INITIALIZING, initializingId));
+        }
+
+        builder.addShard(primaryShard);
+
+        if (primaryShard.relocating()) {
+           // builder.addShard(primaryShard.getTargetRelocatingShard());
```

Comment:
RoutingTable has never the relocation target shard, it's auto-included with the relocating shard. Just remove these 2 lines